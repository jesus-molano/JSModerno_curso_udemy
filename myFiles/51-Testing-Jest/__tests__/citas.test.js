import Citas from "../js/classes/Citas";

describe("Probar la clase de citas", () => {
  const citas = new Citas();
  const id = Date.now();
  test("Agregar nueva cita", () => {
    const citaObj = {
      id,
      mascota: "Alay",
      propietario: "Jesus",
      telefono: "685934281",
      fecha: "06/04/2022",
      hora: "10:30",
      sintomas: "Vomitos",
    };
    citas.agregarCita(citaObj);

    // Prueba
    expect(citas).toMatchSnapshot();
  });

  test("Actualizar cita", () => {
    const citaActualizada = {
      id,
      mascota: "Cathy",
      propietario: "Jesus",
      telefono: "685934281",
      fecha: "06/04/2022",
      hora: "10:30",
      sintomas: "Vomitos",
    };
    citas.editarCita(citaActualizada);

    expect(citas).toMatchSnapshot();
  });

  test("Eliminar cita", () => {
    citas.eliminarCita(id);
    expect(citas).toMatchSnapshot();
  });
});
