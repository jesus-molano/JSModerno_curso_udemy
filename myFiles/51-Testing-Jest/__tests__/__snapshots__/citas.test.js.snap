// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "06/04/2022",
      "hora": "10:30",
      "id": 1624656490655,
      "mascota": "Cathy",
      "propietario": "Jesus",
      "sintomas": "Vomitos",
      "telefono": "685934281",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "06/04/2022",
      "hora": "10:30",
      "id": 1624656490655,
      "mascota": "Alay",
      "propietario": "Jesus",
      "sintomas": "Vomitos",
      "telefono": "685934281",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
